src/applications/disability-benefits/526EZ/tests/actions/actions.unit.spec.js

```diff
@@ -0,0 +1,275 @@
+      const action = setPrestartData(data);
+
+      expect(action.type).to.equal(PRESTART_DATA_SET);
+      expect(action.data).to.equal('2019-04-10T15:12:34.000+00:00');
+    });
+  });
+  describe('resetPrestartState', () => {
+    it('should return action', () => {
+      const action = resetPrestartState();
+
+      expect(action.type).to.equal(PRESTART_STATE_RESET);
+    });
+  });
+  describe('resetPrestartDisplay', () => {
+    it('should return action', () => {
+      const action = resetPrestartDisplay();
+
+      expect(action.type).to.equal(PRESTART_DISPLAY_RESET);
+    });
+  });
+  describe('handleCheckSuccess', () => {
+    it('should return none if no existing ITFs are retrieved', () => {
+      const dispatch = sinon.spy();
+
+      expect(handleCheckSuccess(noData, dispatch)).to.be.true;
+    });
+    it('should return none if no expired or active ITFs are retrieved', () => {
+      const dispatch = sinon.spy();
+
+      expect(handleCheckSuccess(incompleteData, dispatch)).to.be.true;
+    });
+    it('should return expired if retrieved ITFs include expired but not active ITFs', () => {
+      const dispatch = sinon.spy();
+
+      expect(handleCheckSuccess(expiredData, dispatch)).to.be.true;
+      expect(dispatch.calledWith(setPrestartData({ previousExpirationDate: '2016-03-30T16:19:09.000+00:00' }))).to.be.true;
+    });
+    it('should return retrieved if active ITFs are retrieved', () => {
+      const dispatch = sinon.spy();
+
+      expect(handleCheckSuccess(existingData, dispatch)).to.be.false;
+      expect(dispatch.calledWith(setPrestartStatus(PRESTART_STATUSES.succeeded))).to.be.true;
+      expect(dispatch.calledWith(setPrestartData({ currentExpirationDate: '2019-04-10T15:12:34.000+00:00' }))).to.be.true;
+    });
+  });
+  describe('handleCheckFailure', () => {
+    it('should set failed status and return false', () => {
+      const dispatch = sinon.spy();
+
+      expect(handleCheckFailure(dispatch)).to.be.false;
+      expect(dispatch.calledWith(setPrestartStatus(PRESTART_STATUSES.failed))).to.be.true;
+    });
+  });
+  describe('handleSubmitSuccess', () => {
+    it('should dispatch status, and data', () => {
+      const dispatch = sinon.spy();
+
+      handleSubmitSuccess(createdData, dispatch);
+      expect(dispatch.calledWith(setPrestartData({ currentExpirationDate: '2019-04-10T15:12:34.000+00:00' }))).to.be.true;
+      expect(dispatch.calledWith(setPrestartStatus(PRESTART_STATUSES.succeeded))).to.be.true;
+    });
+  });
+  describe('handleSubmitFailure', () => {
+    it('should dispatch error status', () => {
+      const dispatch = sinon.spy();
+
+      handleSubmitFailure(dispatch);
+      expect(dispatch.calledWith(setPrestartStatus(PRESTART_STATUSES.failed))).to.be.true;
+    });
+  });
+  describe('checkITFRequest', () => {
+    afterEach(() => resetFetch());
+    it('should handle success', (done) => {
+      const dispatch = sinon.spy();
+      mockApiRequest({ data: existingData });
```

Comment:
`mockApiRequest()` overrides the `global.fetch`, so we should probably grab `const originalFetch = global.fetch` before the first `describe()` and then put in an `after(() => global.fetch = originalFetch);` in each `describe()` that calls `mockApiRequest()` to avoid any problems in unrelated tests.

Reply:
Thanks @cvalarida, updated.